EduVPN/AppDelegate.swift

```diff
@@ -25,12 +25,16 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         // Insert code here to initialize your application
         UserDefaults.standard.register(defaults: NSDictionary(contentsOf: Bundle.main.url(forResource: "Defaults", withExtension: "plist")!)! as! [String : Any])
         
-        statusItem = NSStatusBar.system().statusItem(withLength: 26)
-        statusItem?.image = #imageLiteral(resourceName: "disconnected-1")
-        statusItem?.menu = statusMenu
- 
+        if #available(OSX 10.12, *) {
+            createStatusItem()
+        }
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(connectionStateChanged(notification:)), name: ConnectionService.stateChanged, object: ServiceContainer.connectionService)
```

Comment:
Don't you need to remove the observer at deinit? Not sure on OSX and AppDelegate though, just something that popped into mind.